Update the README.md file with description, howto and features (#8)
nventive/Reactive.Annex: README.md

﻿# Reactive.Annex

`Reactive.Annex` contains extension methods to help developers make the most out of the Reactive world.
`Reactive.Annex.Uno` also contains an easy-to-use implementation of a dispatcher scheduler following the `IScheduler` interface.

[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](LICENSE)

## Getting Started

Add the `Reactive.Annex` nuget package and its dependencies to your project.

Then, make sure you are using the `System.Reactive.Linq` and `System.Reactive.Concurrency` namespaces in your file.

You are now ready to use these handy extension methods!

If your project uses the [Uno platform](https://platform.uno/), you can also add the `Reactive.Annex.Uno` package to your project. You can now create a `MainDispatcherScheduler`, which implements the `IDispatcherScheduler` interface.

```
var dispatcher = new MainDispatcherScheduler(CoreDispatcher.Main);
```

## Features

`Reactive.Annex` features:

> * `IBackgroundScheduler` and `IDispatcherScheduler`: you no longer need to rely on argument comments to indicate whether the `IScheduler` argument should be a dispatcher or a background scheduler!

Reviewer: ```suggestion
* `IBackgroundScheduler` and `IDispatcherScheduler`: Strongly-typed interfaces to clearly differentiate background schedulers from dispatcher schedulers. No need to rely on variable names anymore
```

Reviewer: What do  you think @jcantin-nventive ?

Author: perfect!

> # Open Source Project Template

Reviewer: ```suggestion
# Reactive.Annex
```

> # Project Title

Reviewer: delete

`Reactive.Annex` contains extension methods to help developers make the most out of the Reactive world.
`Reactive.Annex.Uno` also contains an easy-to-use implementation of a dispatcher scheduler following the `IScheduler` interface.

[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](LICENSE)

## Getting Started

Add the `Reactive.Annex` nuget package and its dependencies to your project.

Then, make sure you are using the `System.Reactive.Linq` and `System.Reactive.Concurrency` namespaces in your file.

You are now ready to use these handy extension methods!

If your project uses the [Uno platform](https://platform.uno/), you can also add the `Reactive.Annex.Uno` package to your project. You can now create a `MainDispatcherScheduler`, which implements the `IDispatcherScheduler` interface.

```
var dispatcher = new MainDispatcherScheduler(CoreDispatcher.Main);
```

## Features

`Reactive.Annex` features:

> * `IBackgroundScheduler` and `IDispatcherScheduler`: you no longer need to rely on argument comments to indicate whether the `IScheduler` argument should be a dispatcher or a background scheduler!

Reviewer: argument comments?

Author: as an example:

https://dev.azure.com/nventive/Umbrella/_git/Umbrella.Analytics?path=%2Fsrc%2FUmbrella.Analytics.Firebase%2FFirebaseAnalyticsService.Android.cs&_a=contents&version=GBmaster

/// <summary>
		/// Initializes a new instance of the <see cref="FirebaseAnalyticsService"/> class.
		/// Android constructor for FirebaseAnalyticsService, needs current provider to log events and activity tracker for user engagements
		/// </summary>
		**/// <param name="backgroundScheduler">Background scheduler</param>
		/// <param name="dispatcherScheduler">Dispatcher scheduler</param>**
		/// <param name="contextProvider">Context provider</param>
		/// <param name="currentActivityProvider">Current activity provider</param>

How would you word it so that it's clearer what I mean?

* `IObservable` extensions
    - `FirstAsync`: Creates a task from an IObservable with the first value observed.
    - `FromAsync`: Converts an async method into an observable sequence.

>     - `SelectManyDisposePrevious`: Runs an asyn action each time your observable sequence produces a new value, while making sure to cancel this action if it's still running as the next value is produced.

Reviewer: ```suggestion
    - `SelectManyDisposePrevious`: Runs an async action each time your observable sequence produces a new value while making sure to cancel the previous action if it's still running.
```
